contracts/TokenContract.sol

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC721/extensions/ERC721EnumerableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/cryptography/draft-EIP712Upgradeable.sol";
+
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC721/utils/ERC721Holder.sol";
+
+import "@dlsl/dev-modules/libs/decimals/DecimalsConverter.sol";
+import "@dlsl/dev-modules/utils/Globals.sol";
+
+import "./interfaces/ITokenFactory.sol";
+import "./interfaces/ITokenContract.sol";
+import "./interfaces/IOwnable.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+
+contract TokenContract is
+    ITokenContract,
+    IOwnable,
+    ERC721EnumerableUpgradeable,
+    EIP712Upgradeable,
+    ReentrancyGuardUpgradeable,
+    ERC721Holder
```

Comment:
We do not need to inherit from **EIP712Upgradeable**, **ReentrancyGuardUpgradeable**, **ERC721Holder** contracts